src/Linux/Host/Broker/Impl/Services/LinuxSystemInfoService.cs

```diff
@@ -135,50 +148,80 @@ public class LinuxSystemInfoService : ISystemInfoService {
 
-            string productPart = "product: ";
-            string vendorPart = "vendor: ";
-            string busPart = "bus info: ";
-
-            // Match anything that looks like 1024M or 1024MB
-            Regex pattern = new Regex("[0-9]+[GM][B]?", RegexOptions.IgnoreCase);
-
-            int count = 1;
-            for (int i=0;  i < lshwDisplayData.Length; ++i) {
-                // find the line that starts with "*-" 
-                if (lshwDisplayData[i].StartsWithIgnoreCase("*-")) {
-                    string productName = GetPart(lshwDisplayData, productPart, ref i);
-                    string vendorName = GetPart(lshwDisplayData, vendorPart, ref i);
-                    string busInfo = GetPart(lshwDisplayData, busPart, ref i);
-                    long vram = 0;
-                    if (!string.IsNullOrEmpty(busInfo)) {
-                        int index = busInfo.IndexOf('@');
-                        if (index >= 0 && index < busInfo.Length) {
-                            string busvalue = busInfo.Substring(index + 1);
-                            for (int j = 0; j < dmesgData.Length; ++j) {
-                                if (dmesgData[i].ContainsIgnoreCase(busvalue)) {
-                                    var match = pattern.Match(dmesgData[i]);
-                                    if (match.Success) {
-                                        vram = GetRamValueMB(match.Value);
-                                        break;
-                                    }
+            foreach (var displayData in lshwDisplayData) {
+                displayData.TryGetValue("product", out string productName);
+                displayData.TryGetValue("vendor", out string vendorName);
+
+                long vram = 0;
+                if (displayData.TryGetValue("bus info", out string busInfo)) {
+                    int index = busInfo.IndexOf('@');
+                    if (index >= 0 && index < busInfo.Length) {
+                        string busvalue = busInfo.Substring(index + 1);
+                        for (int j = 0; j < dmesgData.Length; ++j) {
+                            if (dmesgData[j].ContainsIgnoreCase(busvalue)) {
+                                var match = _ramPattern.Match(dmesgData[j]);
+                                if (match.Success) {
+                                    vram = GetRamValueMB(match.Value);
                                 }
                             }
                         }
                     }
+                }
+                _videoCardInfo.Add(new VideoCardInfo() { VideoCardName = productName, VideoRAM = vram, VideoProcessor = vendorName });
+            }
+        }
+
+        private static IEnumerable<IDictionary<string, string>> ParseDisplayData(IEnumerable<string> lines) {
+            List<Dictionary<string, string>> parsedData = new List<Dictionary<string, string>>();
```

Comment:
This thing will be slightly simpler if, instead of creating a list and filling it, you'll use `yield break` and `yield return` to generate results as you're parsing.